add GET, PUT and DELETE user by id

eviema opened the pull request:
Hi Charlie, please check my homework. Looking forward to your feedback!

YixiuJiang commented on src/main/java/spaceshuttle/controller/UserController.java:
```diff
@@ -46,4 +46,33 @@ User addUser(@RequestBody User newUser) {
+    // Homework 2: PUT user by id
+    @PutMapping(value = "/{id}")
+    public @ResponseBody
+    String updateUserById(@PathVariable Long id,
```
Please use requestbody

eviema commented:
Hi Charlie, 

I've deleted the contents in out/ and added the directory in .gitignore. 

Besides, I have:
- changed return type String to void for methods updateUserById and deleteUserById, to conform to API design guidelines, and
- added getter and setter for user id in user model.

Thanks,
Evie
